README.md

```diff
@@ -8,22 +8,403 @@
+
+```nim
+import chronicles, metrics, metrics/chronicles_support
+
+# ...
+
+info "myCounter", myCounter
+debug "default registry", defaultRegistry
+```
+
+## Testing
+
+When testing, you might want to isolate some collectors by registering them
+into a custom registry:
+
+```nim
+var myRegistry = newRegistry()
+declareCounter myCounter, "help", registry = myRegistry
+echo myRegistry
+
+# this means that `myCounter` is no longer registered in `defaultRegistry`
+echo defaultRegistry
+```
+
+These unoptimised (read "very inefficient") `value()` and `valueByName()`
+procedures for accessing metric values should only be used inside test suites:
+
+```nim
+suite "counter":
+  test "basic":
+    declareCounter myCounter, "help"
+    check myCounter.value == 0
+    myCounter.inc()
+    check myCounter.value == 1
+
+    declareSummary cSummary, "summary with colons in name", name = "foo:bar:baz"
+    cSummary.observe(10)
+    check cSummary.valueByName("foo:bar:baz_count") == 1
+    check cSummary.valueByName("foo:bar:baz_sum") == 10
+```
+
+## Prometheus endpoint
+
+Start an HTTP server listening on 127.0.0.1:9093 from which the Prometheus
+daemon can pull the metrics from all collectors in `defaultRegistry`, plus the
+default ones:
+
+```nim
+startHttpServer()
+```
+
+Or set your own address and port to listen to:
+
+```nim
+import metrics, net
+
+startHttpServer("127.0.0.1", Port(9093))
+```
+
+The HTTP server will run in its own thread. You can open
+"http://127.0.0.1:9093/metrics" in a browser to see what Prometheus sees.
+
+Default metrics available (see also [the relevant Prometheus docs](https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors)):
+
+```text
+process_cpu_seconds_total
+process_open_fds
+process_max_fds
+process_virtual_memory_bytes
+process_resident_memory_bytes
+process_start_time_seconds
+nim_gc_total_mem_bytes
+nim_gc_occupied_mem_bytes
+```
+
+(These `process_*` metrics are only available on Linux, for now.)
+
+## StatsD
+
+Add a [StatsD](https://github.com/statsd/statsd/wiki) export backend where
+metric updates will be pushed as soon as they are created:
+
+```nim
+import metrics, net
+
+when defined(metrics):
+  addExportBackend(
+    metricProtocol = STATSD,
+    netProtocol = UDP,
+    address = "127.0.0.1",
+    port = Port(8125)
+  )
+
+declareCounter myCounter, "some counter"
+myCounter.inc()
+
+# When we incremented the counter, the corresponding data was sent over the wire to the StatsD daemon.
+```
+
+The only supported collector types are counters and gauges. There's a dedicated
+thread that does the networking part. When you update these collectors, data is
+sent over a channel to that thread. If the channel's buffer is full, the data
+is silently dropped. Same for an unreachable backend or any other networking
+error. Reconnections are tried automatically and there's one socket per backend
```

Comment:
Do we drop the latest data or the oldest data? Perhaps the second option would be preferable.

Reply:
Latest, because it's much easier to implement and it doesn't make a difference in practice.